src/mnemonic/mnemonic.py

```diff
@@ -101,16 +101,16 @@ def normalize_string(txt: AnyStr) -> str:
 
     @classmethod
     def detect_language(cls, code: str) -> str:
+        """Scan the Mnemonic until the language becomes unambiguous."""
         code = cls.normalize_string(code)
-        first = code.split(" ")[0]
-        languages = cls.list_languages()
-
-        for lang in languages:
-            mnemo = cls(lang)
-            if first in mnemo.wordlist:
-                return lang
-
-        raise ConfigurationError("Language not detected")
+        possible = set(cls(lang) for lang in cls.list_languages())
+        for word in code.split():
+            possible = set(p for p in possible if word in p.wordlist)
+            if not possible:
+                raise ConfigurationError(f"Language unrecognized for {word!r}")
```

Comment:
typo here `{word!r}`?

Reply:
No, I don't think so -- the purpose is to reveal the source of the Exception -- that 'word' is not recognized in any known language.